src/wx/panel.cpp

```diff
@@ -973,6 +961,307 @@ void GameArea::Resume()
+        // the userdata is freed on disconnect/destruction
+        this->Connect(wxEVT_SIZE,          wxSizeEventHandler(GameArea::OnSize),           NULL, this);
+
+        // We need to check if the buttons stayed pressed when focus the panel.
+        w->Connect(wxEVT_KILL_FOCUS,       wxFocusEventHandler(GameArea::OnKillFocus),     NULL, this);
+
+        // Update mouse last-used timers on mouse events etc..
+        w->Connect(wxEVT_MOTION,           wxMouseEventHandler(GameArea::MouseEvent),      NULL, this);
+        w->Connect(wxEVT_LEFT_DOWN,        wxMouseEventHandler(GameArea::MouseEvent),      NULL, this);
+        w->Connect(wxEVT_RIGHT_DOWN,       wxMouseEventHandler(GameArea::MouseEvent),      NULL, this);
+        w->Connect(wxEVT_MIDDLE_DOWN,      wxMouseEventHandler(GameArea::MouseEvent),      NULL, this);
+        w->Connect(wxEVT_MOUSEWHEEL,       wxMouseEventHandler(GameArea::MouseEvent),      NULL, this);
+
+        w->SetBackgroundStyle(wxBG_STYLE_CUSTOM);
+        w->SetSize(wxSize(basic_width, basic_height));
+
+        if (maxScale)
+            w->SetMaxSize(wxSize(basic_width * maxScale,
+                basic_height * maxScale));
+
+        // if user changed Display/Scale config, this needs to run
+        AdjustMinSize();
+        AdjustSize(false);
+
+        unsigned frame_priority = gopts.retain_aspect ? 0 : 1;
+
+        GetSizer()->Clear();
+
+        // add spacers on top and bottom to center panel vertically
+        // but not on 2.8 which does not handle this correctly
+        if (gopts.retain_aspect)
+#if wxCHECK_VERSION(2, 9, 0)
+            GetSizer()->Add(0, 0, wxEXPAND);
+#else
+            frame_priority = 1;
+#endif
+
+        // this triggers an assertion dialog in <= 3.1.2 in debug mode
+        GetSizer()->Add(w, frame_priority, gopts.retain_aspect ? (wxSHAPED | wxALIGN_CENTER | wxEXPAND) : wxEXPAND);
+
+#if wxCHECK_VERSION(2, 9, 0)
+        if (gopts.retain_aspect)
+            GetSizer()->Add(0, 0, wxEXPAND);
+#endif
+
+        Layout();
+
+#if wxCHECK_VERSION(2, 9, 0)
+        SendSizeEvent();
+#endif
+
+        if (pointer_blanked)
+            w->SetCursor(wxCursor(wxCURSOR_BLANK));
+
+        // set focus to panel
+        w->SetFocus();
+
+        // generate system color maps (after output module init)
+        if (loaded == IMAGE_GBA) utilUpdateSystemColorMaps(gbaLcdFilter);
+        else if (loaded == IMAGE_GB) utilUpdateSystemColorMaps(gbLcdFilter);
+        else utilUpdateSystemColorMaps(false);
+    }
+}
+
+void GameArea::RequestStatusBar(int speed, int frames)
+{
+    wxThreadEvent *event = new wxThreadEvent(WX_THREAD_REQUEST_UPDATESTATUSBAR);
+    event->SetPayload(speed);
+    event->SetExtraLong(frames); // should probably use payload too
+    wxQueueEvent(this, event);
+}
+
+struct PrinterDataDialog {
+    uint16_t* to_print;
+    uint16_t** accum_prdata;
+    int lines, feed;
+    int *accum_prdata_len, *accum_prdata_size;
+};
+PrinterDataDialog printerDataDialog;
```

Comment:
@rkitover YIKES alert here. If you know something that could help me here with wxWidgets, I would appreciate it very much.